Automattic/jetpack
# My Jetpack: Add ConnectedProductOffer component

This PR adds the `<ConnectedProductOffer />` component to `package/my-jetpack` pgk. Basically, it pulls and organizes product data before to provide them to the `<ProductOffer />` component, recently added [here](https://github.com/Automattic/jetpack/issues/23843).

Fixes #

#### Changes proposed in this Pull Request:

* My Jetpack: Add ConnectedProductOffer component

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






* Use Storybook
* Go to the ConnectedProductOffer story
* Switch the product by selecting the slug 

https://user-images.githubusercontent.com/77539/162457966-530c710d-08ca-464a-8965-1245ecbb88e2.mov

* Test My Jetpack interstitial pages,
* Confirm no changes there.
<img width="1662" alt="image" src="https://user-images.githubusercontent.com/77539/162457516-eda2a8c9-6a0f-4d95-ae42-831d45d10405.png">
